source/slang/slang-ir-sccp.cpp

```diff
@@ -676,11 +676,23 @@ struct SCCPContext
     }
     LatticeVal evalLsh(IRType* type, LatticeVal v0, LatticeVal v1)
     {
+        IntInfo info = getIntTypeInfo(type);
+        if (info.isSigned == false)
+        {
+            return evalBinaryIntImpl(
+               type, v0, v1, [](IRUnsignedIntegerValue c0, IRUnsignedIntegerValue c1) { return c0 << c1; });
```

Comment:
I wonder why this doesn't lead to warnings when `evalBinaryIntImpl` calls the lambda with signed integer arguments.

Reply:
There is no precision narrow down or overflow, so in order to turn on this kind of warning, we might need "Wcoercion" in gcc.